util/compare_test_results.py

```diff
@@ -0,0 +1,206 @@
+
+    Returns:
+        tuple: Four lists containing regressions, fixes, newly_skipped, and newly_passing tests
+    """
+    # Find regressions (tests that were passing but now failing)
+    regressions = []
+    for test_path, status in current_flat.items():
+        if status in ("FAIL", "ERROR"):
+            if test_path in reference_flat:
+                if reference_flat[test_path] in ("PASS", "SKIP"):
+
+                    regressions.append(test_path)
+
+    # Find fixes (tests that were failing but now passing)
+    fixes = []
+    for test_path, status in reference_flat.items():
+        if status in ("FAIL", "ERROR"):
+            if test_path in current_flat:
+                if current_flat[test_path] == "PASS":
+                    fixes.append(test_path)
+
+    # Find newly skipped tests (were passing, now skipped)
+    newly_skipped = []
+    for test_path, status in current_flat.items():
+        if (
+            status == "SKIP"
+            and test_path in reference_flat
+            and reference_flat[test_path] == "PASS"
+        ):
+            newly_skipped.append(test_path)
+
+    # Find newly passing tests (were skipped, now passing)
+    newly_passing = []
+    for test_path, status in current_flat.items():
+        if (
+            status == "PASS"
+            and test_path in reference_flat
+            and reference_flat[test_path] == "SKIP"
+        ):
+            newly_passing.append(test_path)
+
+    return regressions, fixes, newly_skipped, newly_passing
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Compare GNU test results and identify regressions and fixes"
+    )
+    parser.add_argument("current_json", help="Path to current run JSON results")
+    parser.add_argument("reference_json", help="Path to reference JSON results")
+    parser.add_argument(
+        "--ignore-file",
+        required=True,
+        help="Path to file with tests to ignore (for intermittent issues)",
+    )
+    parser.add_argument("--output", help="Path to output file for GitHub comment")
+
+    args = parser.parse_args()
+
+    # Load test results
+    try:
+        with open(args.current_json, "r") as f:
+            current_results = json.load(f)
+    except (FileNotFoundError, json.JSONDecodeError) as e:
+        sys.stderr.write(f"Error loading current results: {e}\n")
+        return 1
+
+    try:
+        with open(args.reference_json, "r") as f:
+            reference_results = json.load(f)
+    except (FileNotFoundError, json.JSONDecodeError) as e:
+        sys.stderr.write(f"Error loading reference results: {e}\n")
+        sys.stderr.write("Skipping comparison as reference is not available.\n")
+        return 0
+
+    # Load ignore list (required)
+    if not os.path.exists(args.ignore_file):
+        sys.stderr.write(f"Error: Ignore file {args.ignore_file} does not exist\n")
+        print(f"::error ::Ignore file {args.ignore_file} does not exist")
```

Comment:
Is there some special reason for having (almost) the same error message on both stdout and stderr?
```suggestion
```